docs/explainers.md

```diff
@@ -0,0 +1,152 @@
+| InputCols      | `Array[String]` |         | The names of input columns to the black-box model.                                                           |
+| backgroundData | `DataFrame`     |         | A dataframe containing background data. It must contain all the input columns needed by the black-box model. |
+
+### Vector model explainer params
+
+All vector model explainers ([VectorLIME](#VectorLIME), [VectorSHAP](#VectorSHAP)) support the following params:
+
+| Param          | Type        | Default | Description                                                                                                    |
+|----------------|-------------|---------|----------------------------------------------------------------------------------------------------------------|
+| InputCol       | `String`    |         | The names of input vector column to the black-box model.                                                       |
+| backgroundData | `DataFrame` |         | A dataframe containing background data. It must contain the input vector column needed by the black-box model. |
+
+### Image model explainer params
+
+All image model explainers ([ImageLIME](#ImageLIME), [ImageSHAP](#ImageSHAP)) support the following params:
+
+| Param         | Type     | Default       | Description                                                       |
+|---------------|----------|---------------|-------------------------------------------------------------------|
+| InputCol      | `String` |               | The names of input image column to the black-box model.           |
+| CellSize      | `Double` | 16            | Number that controls the size of the superpixels.                 |
+| Modifier      | `Double` | 130           | Controls the trade-off spatial and color distance of superpixels. |
+| SuperpixelCol | `String` | "superpixels" | The column holding the superpixel decompositions.                 |
+
+### Text model explainer params
+
+All text model explainers ([TextLIME](#TextLIME), [TextSHAP](#TextSHAP)) support the following params:
+
+| Param     | Type     | Default  | Description                                            |
+|-----------|----------|----------|--------------------------------------------------------|
+| InputCol  | `String` |          | The names of input text column to the black-box model. |
+| TokensCol | `String` | "tokens" | The column holding the text tokens.                    |
+
+### `TabularLIME`
+
+| Param               | Type            | Default     | Description                                                          |
+|---------------------|-----------------|-------------|----------------------------------------------------------------------|
+| CategoricalFeatures | `Array[String]` | empty array | The name of columns that should be treated as categorical variables. |
```

Comment:
Confused why LIME would have this but not SHAP

Reply:
For tabular data, LIME treats categorical features and numerical features differently. For categorical features, it samples based on the value distribution from the background dataset, for numerical features, it samples from a Guassian distribution.

Kernel SHAP treats these features in the same way: create sample coalitions, and sample from background dataset if a feature is "turned off".

Reply:
thanks for that detail! Can we put some of that in this or a future doc as it was very helpful

Reply:
great idea. I'll update the doc before merging.

Reply:
done.